typing/typecore.ml

```diff
@@ -1715,38 +1725,6 @@ and type_pat_aux
             pat_attributes = [];
             pat_env = !env }
       end
-  | Ppat_constraint(
-      {ppat_desc=Ppat_var name; ppat_loc=lloc; ppat_attributes = attrs},
-      ({ptyp_desc=Ptyp_poly _} as sty)) ->
-      (* explicitly polymorphic type *)
-      assert construction_not_used_in_counterexamples;
-      let type_mode =
-        if has_local_attr_pat sp then Alloc_mode.Local
-        else Alloc_mode.Global
-      in
-      let cty, ty, force =
-        Typetexp.transl_simple_type_delayed !env type_mode sty
-      in
-      unify_pat_types ~refine lloc env ty (instance expected_ty);
-      pattern_force := force :: !pattern_force;
-      begin match ty.desc with
-      | Tpoly (body, tyl) ->
-          begin_def ();
-          init_def generic_level;
-          let _, ty' = instance_poly ~keep_names:true false tyl body in
-          end_def ();
-          let id = enter_variable lloc name alloc_mode.mode ty' attrs in
-          rvp k {
-            pat_desc = Tpat_var (id, name);
-            pat_loc = lloc;
-            pat_extra = [Tpat_constraint cty, loc, sp.ppat_attributes];
-            pat_type = ty;
-            pat_mode = alloc_mode.mode;
-            pat_attributes = [];
-            pat_env = !env
-          }
-      | _ -> assert false
-      end
```

Comment:
Why is this case no longer needed?

Reply:
Now handled by the general `Ppat_constraint` case